R/students.R

```diff
@@ -311,12 +329,13 @@ s_03 <- function(input_df=usheUtils::fake_student_df) {
   output_df <- input_df %>%
     # Calculate intermediate fields
     # Note: ssn that start with 9 are actually IRS issued tax IDs; USHE only wants valid ssn.
-    mutate( s_id = if_else(is.na(ssn) |
-                          str_starts(ssn, "9") |
-                          str_length(ssn) >= 10 |
-                          str_length(ssn) <= 8,
+    mutate(s_id_intermediate = str_replace_all(ssn, "-", "")) %>%
+    mutate( s_id = if_else(is.na(s_id_intermediate) |
+                          str_starts(s_id_intermediate, "9") |
+                          str_length(s_id_intermediate) >= 10 |
+                          str_length(s_id_intermediate) <= 8,
                           paste0('D', student_id),
-                          str_replace_all(ssn, "-", "") ) )  %>%
+                          s_id_intermediate ) )  %>%
 
```

Comment:
@cdemkegit Given the complexity this function is needing. I think it would be a good idea to merge s_03 and s_04 at this point, since the logic needs to line up between the two.

I'm not sure what that merge would look like exactly. But I just wanted to make note of that here.